core/src/avm2/globals/function.rs

```diff
@@ -15,8 +16,16 @@ use gc_arena::GcCell;
 pub fn instance_init<'gc>(
     activation: &mut Activation<'_, 'gc>,
     this: Object<'gc>,
-    _args: &[Value<'gc>],
+    args: &[Value<'gc>],
 ) -> Result<Value<'gc>, Error<'gc>> {
+    if !args.is_empty() {
```

Comment:
Function also has a call handler that just forwards to `new Function(...)` (`Function(...)` is equivalent to `new Function(...)`)- could you add that in this PR? It might fix some tests.